### PR TITLE
[tools] Introduce github release tool

### PR DESCRIPTION
**Pull request by mhs4670go**
This draft is for automation of github release.

e.g.
```shell
# 1
$ ./git_release.sh --tag 1.9.0 --commitish 1.9.0 --token af353sdfljk354l5k4kl5 \
--repo_owner mhs4670go --repo_name test_repo --release_note local/repo/release_note \
--asset test.sh --asset test2.sh

# 2
$ ./git_release.sh --tag 1.10.0 --commitish master --token slkfjsdlf32443kifslk3 \
--repo_owner company --repo_name internal_repo --release_note local/repo/release_note \
--release_name "internal release!!" --draft \
--host_name github.sec.company.net
```

Related : #3832 
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

File: infra/scripts/git_release.sh [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+# Check if the release already exists
+RELEASE_URL=$(curl -s --request GET --header "Authorization: token ${USER_TOKEN}" \
+https://${HOST_NAME}/repos/${REPO_OWNER}/${REPO_NAME}/releases/tags/${TAG_NAME} | \
+jq -r '.url')
+
+if [ $RELEASE_URL != null ]; then
+  echo "[ERROR] The tag name you specified already exists."
+  exit 0
```
Comment (mhs4670go) [inline infra/scripts/git_release.sh:176]:
```suggestion
  exit 2
```

File: infra/scripts/git_release.sh [hunk 2]
```diff
@@ -0,0 +1,194 @@
+#!/bin/bash
+# This script is to automate the process of montly release with github api
```
Comment (seanshpark) [inline infra/scripts/git_release.sh:2]: typo `monthly`
Let's use `API` capital letters.

File: infra/scripts/git_release.sh [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+function Usage()
+{
+  echo "Usage: ./$(basename ${BASH_SOURCE[0]}) --tag TAG --release_note RELEASE_NOTE \
+--token TOKEN [--release_name RELEASE_NAME] [--commitish COMMITISH] [--draft] \
+[--host_name HOST_NAME] [--repo_owner REPO_OWNER] [--repo_name REPO_NAME] [--asset] ..."
+  echo ""
+  echo "[OPTIONS]"
+  echo "--tag              The name of the tag"
+  echo "--release_name     The name of the release"
```
Comment (seanshpark) [inline infra/scripts/git_release.sh:21]: (optional) it would help if example is added

File: infra/scripts/git_release.sh [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+  echo "--release_note     Path of text file describing the contents of the release"
+  echo "--commitish        The commitish value that determines where the Git tag is created from"
+  echo "--draft            Create a draft release"
+  echo "--token            User token for authentication"
```
Comment (seanshpark) [inline infra/scripts/git_release.sh:25]: `commitish` and `token` seems to be related with github API. If possible could you please add some comments about where to reference about how to get these valuses? describing steps would be wonderful.

**Comment by seanshpark** [comment]
Thanks!

**Comment by mhs4670go** [comment]
@seanshpark @hseok-oh PTAL:)

**Review by seanshpark — APPROVED** [review]
LGTM

File: tools/release_tool/git_release.sh [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+  echo "--commitish        The commitish value that determines where the Git tag is created from"
+  echo "--draft            Create a draft release"
+  echo "--token            User token for authentication"
+  echo "--host_name        Host name for endpoint URL [Enterprise-specific endpoint only]"
+  echo "--repo_owner       Owner of the repository"
+  echo "--repo_name        The name of the repository"
+  echo "--asset            Path of release asset"
+  echo ""
+  echo "[EXAMPLE]"
+  echo "$ ./git_release.sh --tag 1.9.0 --commitish release/1.9.0 --token 0de25f1ca5d1d758fe877b18c06 \\"
```
Comment (hseok-oh) [inline tools/release_tool/git_release.sh:30]: Please hide token. (regardless it is not valid)

Comment (hseok-oh) [reply tools/release_tool/git_release.sh:30]: @mhs4670go Is this token invalid? If it is invalid token, I'll approve this PR.

Comment (mhs4670go) [reply tools/release_tool/git_release.sh:30]: @hseok-oh Yes:) That is dummy token.

**Review by hseok-oh — APPROVED** [review]
LGTM

**Pull request merged**